### PR TITLE
Datamanager feature: update(key, object, forceCreate)

### PR DESCRIPTION
**Pull request by UnFlimFlammable**
Please do not update the README or Change Log, we will do this when we merge your PR.

This PR:
Adds a new feature

Describe the changes below:
This adds an update method that will take an object and update the existing object in the datamanager with the fields in the object passed in. 

Example:

```
this.data.set('test', {x:10, y:20});
this.data.update('test', {x:20});

var newTest = this.data.get('test');
//newTest.x = 20, newTest.y = 20
```

### CODE REVIEW

**Comment by KingCosmic** [comment]
I for one would like this to be added 💃

**Comment by samme** [comment]
I think this needs `eslint --fix`.

File: src/data/DataManager.js [hunk 1]
```diff
@@ -378,6 +378,38 @@ var DataManager = new Class({
...
+     */
+    update: function(key, object, forceCreate)
+    {
+        var existing = this.get(key);
+        if(forceCreate === undefined){forceCreate = true}
+        //If the object already exists
+        if(existing !== undefined){
+            for(var key in object){
+                //If the field exists inside the object, update, otherwise if forceCreate is true add field anyway
+                if(existing[key] !== null || forceCreate === true){
```
Comment (samme) [inline src/data/DataManager.js]:
```suggestion
                if ((key in existing) || forceCreate === true) {
```

**Comment by photonstorm** [comment]
eslint issues aside, you can already do this:

```
this.data.set('test', {x:10, y:20});

this.data.values.test.x = 20;

var newTest = this.data.get('test');
//newTest.x = 20, newTest.y = 20
```

**Comment by jcyuan** [comment]
datamanager has `frozen` feature, you can't do this by a `update` method.

**Comment by photonstorm** [comment]
Thanks for this PR, however, I'm afraid I'm going to close it off, as it doesn't pass the eslint check and you can already do this via the `values` property of the Data Manager.

**Pull request closed without merging**